### PR TITLE
Fix up hashes, cleanup validation, some refactoring

### PR DESCRIPTION
**Pull request by Bowbaq**
@buzztroll made a few tweaks to make things work better on my end, feel free to merge into your branch if you're happy with them

### CODE REVIEW

File: builtin/providers/azurerm/resource_arm_simple_lb.go [hunk 1]
```diff
@@ -166,7 +167,7 @@ func resourceArmSimpleLb() *schema.Resource {
 func resourceARMLoadBalancerRuleHash(v interface{}) int {
 	m := v.(map[string]interface{})
 	rule := fmt.Sprintf("%s-%s-%s-%s-%d-%d", m["name"].(string), m["protocol"].(string), m["probe_name"].(string), m["load_distribution"].(string), m["frontend_port"], m["backend_port"])
-	return hashcode.String(rule)
+	return hashcode.String(strings.ToLower(rule))
```
Comment (Bowbaq) [inline builtin/providers/azurerm/resource_arm_simple_lb.go:170]: Occasional case differences were triggering bogus diffs, this fixed it

File: builtin/providers/azurerm/resource_arm_simple_lb.go [hunk 2]
```diff
@@ -175,153 +176,119 @@ func resourceARMLoadBalancerProbeHash(v interface{}) int {
...
-	frontendIpId := *(*loadBalancer.Properties.FrontendIPConfigurations)[0].ID
-
-	// then; the subnets:
+	log.Printf("[resourceArmSimpleLb] pullOutLbRules %v", *loadBalancer.Properties.Probes)
 	outRules := []network.LoadBalancingRule{}
 
-	log.Printf("[resourceArmSimpleLb] pullOutLbRules will use frontend %s and backend %s", frontendIpId, backendPoolId)
-	rules := d.Get("rule").(*schema.Set)
-
-	if rules.Len() > 0 {
```
Comment (Bowbaq) [inline builtin/providers/azurerm/resource_arm_simple_lb.go:277]: I reduced the nesting a bit here

File: builtin/providers/azurerm/resource_arm_network_interface_card.go [hunk 3]
```diff
@@ -344,6 +344,12 @@ func resourceArmNetworkInterfaceIpConfigurationHash(v interface{}) int {
 	if m["public_ip_address_id"] != nil {
 		buf.WriteString(fmt.Sprintf("%s-", m["public_ip_address_id"].(string)))
 	}
+	if m["load_balancer_backend_address_pools_ids"] != nil {
```
Comment (Bowbaq) [inline builtin/providers/azurerm/resource_arm_network_interface_card.go:347]: This makes sure the NIC gets updated properly if the LB changes

**Comment by buzztroll** [comment]
This looks good to me.  I am running the tests then I will merge.  Thanks!   Would you mind adding verbose description to the commit message about what bug/issue is being fixed for the NIC?

**Comment by Bowbaq** [comment]
@buzztroll looks like I saw that too late :/

**Pull request merged**